src/main/java/net/neoforged/neoforge/event/entity/item/ObtainItemEvent.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.item;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.item.ItemEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.Projectile;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.Event;
+import net.neoforged.neoforge.event.EventHooks;
+import net.neoforged.neoforge.event.entity.player.ItemEntityPickupEvent;
+
+/**
+ * Fired when player gets an item from sources list below. Note that other means of giving item to players / other entities are not accounted for.<br>
+ * For item pickups, Called after {@link ItemEntityPickupEvent.Pre} and before {@link ItemEntityPickupEvent.Post}.<br>
+ * Listeners of this event is allowed to reduce the amount of this stack, but is not allowed increase or change data component of this stack.<br>
+ * Mods that want to give items to player should call {@link EventHooks#addToInventory(Player, ItemStack, Source)}<br>
+ * Mods that want entity to pick up items should call {@link EventHooks#onObtainItem(LivingEntity, ItemStack, Source)}<br>
+ * <br>
+ * Default Sources:
+ * <li>ItemEntitySource for item entity pickup</li>
+ * <li>ProjectileSource for arrow and trident</li>
+ * <li>ItemContainerReturn for container return when consuming items (Not being used by NeoForge yet)</li>
```

Comment:
The `(Not being used by NeoForge yet)` probably should be removed. Or do you mean drinking the last honey bottle will not use this source right now?

Reply:
Thanks. Should I retarget this PR to 1.21.3?

Reply:
The PR is already targeting 1.21.3 branch. After this is merged, you can do a backport PR to 1.21.1